source/mge.js

```diff
@@ -0,0 +1,988 @@
+                if (typeof this.type === "string") {return this.type.split(".").slice(-1) + '(' + content + ')';}
+                return this.type.__name__ + '(' + content + ')';
+            }
+            formatString() {
+                var type = typeof this.type !== "string" ? this.type.__name__ : this.type.split(".").slice(-1)[0];
+                if (type === "tuple") {
+                    let content = "";
+                    for (var child of this.children_defs) {
+                        content += child.formatString() + ',';
+                    }
+                    return "(" + content + ")";
+                }
+                if (type === "slice") {
+                    return this.children_defs[0].formatString() + ":" + this.children_defs[1].formatString() + ":" + this.children_defs[2].formatString();
+                }
+                if (type === "list") {
+                    let content = "";
+                    for (const child of this.children_defs) {
+                        content += child.formatString() + ',';
+                    }
+                    return "[" + content + "]";
+                }
+                if (type === "dict") {
+                    var content = "";
+                    for (var i = 0; i < this.children_defs.length; i++) {
+                        content += this.aux_data[i] + ":" + this.children_defs[i].formatString;
+                    }
+                    return "{" + content + "}";
+                }
+                return "unknown";
+            }
+        });
+        this.registerType('megengine.traced_module.pytree.LeafDef', class {
+            toString() {
+                let content = '';
+                if (this.const_val !== null) {content += this.const_val;}
+                else {content += '[';}
+                for (var t of Object.values(this.type)) {
+                    content += t.__name__;
+                }
+                content += ']';
+                return content;
+            }
+
+            formatString() {
+                if (this.const_val !== null) {return this.const_val;}
+                else
+                if (this.type[0].__module__ !== undefined) {return this.type[0].__name__;}
+                return "None";
+            }
+        });
+        this.registerType('megengine.tensor.Tensor', class {
+            constructor(data, dtype, device) {
+                this.data = data;
+                this.dtype = dtype;
+                this.device = device;
+            }
+        });
+
+        this.registerType('megengine.core.tensor.dtype.QuantDtypeMeta', class {
+            constructor(name, cname, np_dtype, qmin, qmax, is_signed) {
+                this.name = name;
+                this.cname = cname;
+                this.np_dtype = np_dtype;
+                this.qmin = qmin;
+                this.qmax = qmax;
+                this.is_signed = is_signed;
+            }
+        });
+
+    }
+};
+
+megengine.Container = class {
```

Comment:
Merge `megengine.Container` and `megengine.Container.Pickle` into a single `megengine.Pickle` class.

Reply:
done